### PR TITLE
Fix NULL-deref in combobox with NULL items

### LINKED ISSUE #226: Bug: nk_console_combobox NULL-derefs on NULL items string
`nk_console_combobox_update` (nuklear_console_combobox.h:123) iterates `items_separated_by_separator[i]` with no NULL guard. Calling `nk_console_combobox(parent, "X", NULL, '|', &sel)` — or invoking `nk_console_combobox_update` with NULL items — will segfault.

The outer `nk_console_combobox` constructor at line 167 also passes the argument straight through without a check.

## Suggested Implementation
Early-return with just the back button (and clamp `*selected = 0`) when `items_separated_by_separator == NULL`, mirroring the `!combobox || !combobox->data` guard already at the top of `nk_console_combobox_update`.

## QA
- Add a test that constructs a combobox with NULL items and asserts no crash, `*selected == 0`, and zero data children.
- Confirm existing combobox tests still pass.

### PR DESCRIPTION
**Pull request by RobLoach**
Guard `nk_console_combobox_update` against a NULL `items_separated_by_separator` by early-returning after adding the back button and clamping `*selected` to 0, mirroring the existing `!combobox || !combobox->data` guard. A test is added that constructs a combobox with NULL items and asserts no crash, `*selected == 0`, and exactly one child (the back button).

Fixes #226

### CODE REVIEW

File: test/nuklear_console_test.c [hunk 1]
```diff
@@ -228,6 +228,16 @@ int main() {
...
+    // nk_console_combobox() with NULL items must not crash
+    {
+        int value_null_combobox = 5;
+        nk_console* combobox_null = nk_console_combobox(console, "NullComboBox", NULL, '|', &value_null_combobox);
+        assert(combobox_null != NULL);
+        assert(value_null_combobox == 0);
+        // Only the back button should be a child (no item children).
+        assert(cvector_size(combobox_null->children) == 1);
+    }
+
```
Comment (RobLoach) [inline test/nuklear_console_test.c]:
``` suggestion
```
